bin/propolis-server/src/lib/server.rs

```diff
@@ -953,6 +955,77 @@ async fn instance_issue_crucible_snapshot_request(
     Ok(HttpResponseOk(()))
 }
 
+/// Issues a volume_construction_request replace to a crucible backend.
+#[endpoint {
+    method = PUT,
+    path = "/instance/disk/{id}/vcr",
+}]
+async fn instance_issue_crucible_vcr_request(
+    rqctx: RequestContext<Arc<DropshotEndpointContext>>,
+    path_params: Path<api::VCRRequestPathParams>,
+    request: TypedBody<api::InstanceVCRReplace>,
+) -> Result<HttpResponseOk<()>, HttpError> {
+    let path_params = path_params.into_inner();
+    let request = request.into_inner();
+    let new_vcr_json = request.vcr_json;
+    let disk_name = request.name;
+    let log = rqctx.log.clone();
+
+    // Get the instance spec for storage backend from the disk name.  We use
+    // the VCR stored there to send to crucible along with the new VCR we want
+    // to replace it.
+    let vm_controller = rqctx.context().vm().await?;
+    let mut spec = vm_controller.instance_spec().await;
+    let VersionedInstanceSpec::V0(v0_spec) = &mut *spec;
+
+    let (readonly, old_vcr_json) = {
+        let bes = &v0_spec.backends.storage_backends.get(&disk_name);
+        if let Some(StorageBackendV0::Crucible(bes)) = bes {
+            (bes.readonly, &bes.request_json)
+        } else {
+            let s = format!("Crucible backend for {:?} not found", disk_name);
+            return Err(HttpError::for_not_found(Some(s.clone()), s));
+        }
+    };
+
+    // Get the crucible backend so we can call the replacement method on it.
+    let crucible_backends = vm_controller.crucible_backends();
+    let backend = crucible_backends.get(&path_params.id).ok_or_else(|| {
+        let s = format!("No crucible backend for id {}", path_params.id);
+        HttpError::for_not_found(Some(s.clone()), s)
+    })?;
+
+    slog::info!(
+        log,
+        "{:?} {:?} replace {:?} with {:?}",
+        disk_name,
+        path_params.id,
+        old_vcr_json,
+        new_vcr_json,
+    );
+
+    // Try the replacement.
+    // Crucible does the heavy lifting here to verify that the old/new
+    // VCRs are different in just the correct way and will return error
+    // if there is any mismatch.
+    backend.vcr_replace(old_vcr_json, &new_vcr_json).await.map_err(|e| {
+        HttpError::for_bad_request(Some(e.to_string()), e.to_string())
+    })?;
+
+    // Our replacement request was accepted.  We now need to update the
+    // spec stored in propolis so it matches what the downstairs now has.
+    let new_storage_backend: StorageBackendV0 =
+        StorageBackendV0::Crucible(CrucibleStorageBackend {
+            readonly,
+            request_json: new_vcr_json,
+        });
+    v0_spec.backends.storage_backends.insert(disk_name, new_storage_backend);
+
+    slog::info!(log, "Replaced the VCR in backend of {:?}", path_params.id);
```

Comment:
nit: I might push this kind of operational log down into `vcr_replace`, but it's up to you.

Reply:
Makes sense to me. I suspect we'll want something similar in whatever we ultimately do for #205.